### DISCUSSION TITLE
When to use OCSMesh

### DISCUSSION THREAD

**Discussion by acroucher** [noaa-ocs-modeling/OCSMesh#36] [General]
Are there any particular situations for which it would not be appropriate to use OCSMesh (or other data-driven automatic meshing tools, e.g. OceanMesh2D) to create your mesh?

Background: I'm starting to create a new SCHISM model of a moderately complex domain containing a lot of islands and some intertidal zones. I've had SMS recommended to me for creating the mesh. The more manual, hands-on graphical approach of SMS I guess has a lot of flexibility for custom meshing to handle difficult situations. However, I am drawn to trying the data-driven, script-based approach offered by tools like OCSMesh (especially as it's also open-source).

So, are there any particular situations where you think a more manual tool like SMS might be necessary?

**Comment by SorooshMani-NOAA** [comment]
I think the fact that you have more manual control over the mesh in SMS and  because it has a lot of algorithms and partial automations developed over years, it can give you better final mesh results, but this is at the cost of manual tweaking and also might not be easily reproducible or portable from one region to another. I don't know if there has ever been a head-to-head comparison between SMS and other mesh generation tools for a more quantitative or technical opinion. @WPringle what do you think?

**Comment by WPringle** [comment]
@SorooshMani-NOAA I don't think it is really necessary to use SMS except that for instance we don't have quad elements setup in OCSMesh or OceanMesh2D yet. So if you need to add these to the mesh, for now can use SMS, but we also want to add these to our scripting meshers. The other thing is if you want to do fine tuning of the mesh in post-processing stage, to align some points along certain features where it wasn't easy to provide up front in the data for the scripting approach. But I still think most things are possible to do by scripting, just may not always be easy to implement at first.

**Reply by acroucher** [reply]
The SMS documentation recommends using quad elements (if the model supports them, e.g. SCHISM) for some situations, e.g. representing channels. Do you think there are circumstances in which this is really necessary, i.e. good results cannot be obtained using triangular elements?

**Reply by WPringle** [reply]
I guess Joseph Zhang would know more about that, but quads are useful for rivers and channels because it efficiently discretizes along the flow path - can be elongated along-flow and short across-flow.

**Reply by acroucher** [reply]
Yes, although naively I'd have thought anything you can do with a quad you can probably also do with the same quad divided into two triangles - and I understand SCHISM is less fussy about triangle quality than quad quality.

**Comment by krober10nd** [comment]
The challenges William and I faced in developing regional scale meshes for various projects with SMS was the main motivation behind developing OceanMesh2D. These problems have been well-enumerated in perhaps dozens of talks and presentations that appear when you research the software over the last half-dozen years. 

In engineering problems, there are new developments coming to OceanMesh2D that enable faithful placement of high-aspect ratio structures.
